fix(linter): do not output number of rules with nested configs

camchenry opened the pull request:
This is a small output update to fix some noisy snapshot tests that output the number of rules. When using nested configs, outputting the number of rules is not helpful I think since it varies from file to file, but open to feedback on this.

camchenry commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9476?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9476** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9476?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9476?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9475** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9475?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9473** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9473?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9472** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9472?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9308** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9308?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9217** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9217?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9156** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9156?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9155** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9155?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9153** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9153?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9152** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9152?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9476).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

shulaoda commented on crates/oxc_linter/src/lib.rs:
```diff
@@ -106,8 +106,11 @@ impl Linter {
+    /// nested configurations in use, in which case it returns `None` since the
+    /// number of rules depends on which file is being linted.
+    pub fn number_of_rules(&self) -> Option<usize> {
+        if self.nested_configs.is_empty() { Some(self.config.number_of_rules()) } else { None }
```
`self.nested_configs.is_empty().then_some(self.config.number_of_rules())` ?

Not blocking the PR, just noticed it and thought of this syntactic sugar.

camchenry replied:
nice, don't think I remembered this API existed.

Sysix reviewed (approved):
The number of rules is already not the exact representation of used rules.
Remapped rules are not count as separate ones and probably the count for `overrides` is ignored.
It is part of the discussion: https://github.com/oxc-project/oxc/discussions/9421

I am thinking about removing this information in general

camchenry commented:
### Merge activity

* **Mar 3, 10:49 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
fix(linter): do not output number of rules with nested configs (#9476)

This is a small output update to fix some noisy snapshot tests that output the number of rules. When using nested configs, outputting the number of rules is not helpful I think since it varies from file to file, but open to feedback on this.
